src/libs/antares/study/area/list.cpp

```diff
@@ -909,13 +909,29 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
             ret = area.hydro.series->loadFromFolder(study, area.id, buffer) && ret;
         }
 
+        if (area.hydro.series && study.header.version < 870)
+        {
+            std::shared_ptr<DataTransfer> datatransfer = std::make_shared<DataTransfer>();
+            buffer.clear() << study.folderInput << SEP << "hydro";
+
+            datatransfer->LoadFromFolder(study, buffer, area);
+            datatransfer->SupportForOldStudies(study, buffer, area);
+        }
+
         if (area.hydro.series)
         {
             buffer.clear() << study.folderInput << SEP << "hydro" << SEP << "series";
-            ret = area.hydro.series->SupportForOldStudies(study, area.id, buffer) && ret;
             ret = area.hydro.series->LoadHydroPowerCredits(study, area.id, buffer) && ret;
         }
 
+        if (study.header.version >= 870 && !area.hydro.isDeleted)
+        {
```

Comment:
This piece of code will be executed after upgrading a study and it will be executed every time data is loading. Idea is to this piece of code be executed just once after upgrading a study. Once file is deleted we don't need execution of this piece of code anymore. This implementation is not working properly, so this is next to be figured out.

Reply:
Done!